File: Assets/Our Assets/Mergeable/C Sharp/PlayerHealth.cs
/// <summary>
/// PlayerHealth.cs
/// Oct 20, 2010
/// Peter Laliberte
/// 
/// Modified By Calvin Slusarski
/// February 19 2011
/// Display the players health in game
/// 
/// Attach this class to your player character
/// </summary>
using UnityEngine;
using System.Collections;

public class PlayerHealth : MonoBehaviour {
	public int maxHealth = 100;
	public int curHealth = 100;
	// Added By Calvin: Adjust colors via inspector
	//public GUIStyle newStyles = new  GUIStyle();
	
	public float healthBarLength;

	// Use this for initialization
	void Start () {
		healthBarLength = Screen.width / 2;
	}
	
	// Update is called once per frame
	void Update () {
		AddjustCurrentHealth(0);
	}
	
	void OnGUI() {
		GUI.Box(new Rect(10, 10, healthBarLength, 20), curHealth + "/" + maxHealth/*, newStyles*/);
	}
	
	public void AddjustCurrentHealth(int adj) {
		curHealth += adj;
		
		if(curHealth < 0)
			curHealth = 0;
		
		if(curHealth > maxHealth)
			curHealth = maxHealth;
		
		if(maxHealth < 1)
			maxHealth = 1;
		
		healthBarLength = (Screen.width / 2) * (curHealth / (float)maxHealth);
	}
}

File: Assets/Our Assets/Mergeable/JavaScript/GUI/PlayerHealth.js
/// <summary>
/// PlayerHealth.js
/// Modified By Calvin Slusarski
/// February 19 2011
/// Display the players health in game
/// Attach this class to your player character
/// </summary>
/// PlayerHealth.cs
/// Oct 20, 2010
/// Peter Laliberte

	var characterStats : Stats;
	var healthBarLength = 0;
	var textureForBehindText : Texture;
	
	function Start() {
		healthBarLength = Screen.width /4;
	}
		
	function OnGUI() {
		currentBarLength = healthBarLength * characterStats.health / characterStats.maxHealth;
//		GUI.Box(Rect(10, 10, currentBarLength, 20), characterStats.health + "/" + characterStats.maxHealth);
		shape = Rect(10, 10, currentBarLength, 20);
//		content = GUIContent (characterStats.health + "/" + characterStats.maxHealth, textureForBehindText);
		content = GUIContent (textureForBehindText);
		GUI.Box(shape, content);
		GUI.Box(Rect(10, 30, 60, 25), characterStats.eleType);
//		GUIContent (text : string, image : Texture) :
	}
